src/parser.go

```diff
@@ -17,29 +17,42 @@ type Parser interface {
 
 type ZshParser struct {
 	path string
+	batchSize int64
 }
 
 func (p ZshParser) Parse() string {
-	line := readSecondToLastLine(p.path)
+	line := readSecondToLastLine(p.path, p.batchSize)
 	splits := strings.Split(line, ";")
 	return splits[len(splits) - 1]
 }
 
 func NewZshParser(p string) Parser {
-	return ZshParser{path: p}
+	return ZshParser{path: p, batchSize: defaultBatchSizeKb}
+}
+
+// NewZshParserWithBatchSize creates a zsh parser with custom batchSize.
+func NewZshParserWithBatchSize(p string, batchSize int64) Parser {
+	return ZshParser{path: p, batchSize: batchSize}
 }
 
 // Helpers
 
-func readSecondToLastLine(filePath string) string {
+func readSecondToLastLine(filePath string, batchSize int64) string {
 	file, err := os.Open(filePath)
 	Check(err)
 	defer file.Close()
 
-	buf := make([]byte, batchSizeKb)
 	stat, err := os.Stat(filePath)
 	Check(err)
-	start := stat.Size() - batchSizeKb
+
+	size := stat.Size()
```

Comment:
Isn't there a better way to express this whole part?
```
start := max(size - batchSize, 0)
buf := make([]byte, min(size, batchSize))
```

Reply:
Hahaha, there is no max / min function for integers, _but it's easy to implement your own_ 😂 
https://yourbasic.org/golang/max-min-function/

Reply:
https://github.com/tome-sh/tome/pull/1/commits/474f18e6491981065a1fb5edd5cceddab625d0ce